lambda_functions/archive_service/post_comment_on_item.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import json
+from core_layer import helper
+from core_layer.db_handler import Session
+from core_layer.handler import comment_handler, item_handler
+
+
+def post_comment_on_item(event, context = None):
```

Comment:
Couldn't we just create a general Lambda handler that creates all types of comments (item, review, ...)?

Reply:
In the review process, I would leave the comment in the general update_review function. 
For the cases in which we add comments in a separate lambda, I like the idea of having one function for all types of comments, but currently we only have one usecase for this. So maybe we can leave it as it is and make this function more general, once we need it?

Reply:
Agree on the second part.

For the first part, is the intention, that a user submits a comment on an item directly alongside their review?
If yes, then I'm fine with it - but we should then look into the order of method calls there.
If I remember correctly, the comment was written to DB before some checks were executed and the review is written to the DB.

Reply:
You are right, the comment_handler is called before the review is written to db. 
The handler function also contains a `session.commit()`. Couldn't we use a context manager (`with Session() as session`) and omit the commit in the handler?

Reply:
Using the context manager with Session() as session will only initiate and close a session.
That means, there is no implicit transaction handling as in the construction using` session.beginn()`
```
with session() as session
    with session.begin()
        ...
```
We decided not to use the transaction handling with implicit commit and rollback - due to negative side effects, see comments [here](https://docs.google.com/document/d/1vCl_9v3x5DgniAOmC8cp5gk1zbRgw986/edit).
Most crucial point is the exception handling. 
Actually we are working with explicit commits in the handler functions - but they should be organizied in a way that there is one appropriate transaction, not multiple small commits.
If different handlers are involved you have to take care of the correct order - first updating a review, than creating a comment.
Have some thoughts how to deal with this situation in general for later discussion.

Reply:
I moved the handler call to the "update" part of the function as discussed. Thanks again for the comprehensive explanation.

Reply:
Calling comment_handler.create_comment() **after** successfully updating review answers in db seems to be the correct order.